fix: Ambiguity in VectorTestBase::makeFlatVector with brace initialization

Super-long opened the pull request:
## Problem
When using `makeFlatVector` with brace initialization syntax like `makeFlatVector<int64_t>({803})`, there was a compilation ambiguity between the existing overloads:
- `makeFlatVector(vector_size_t size, ...)` - where `size_t` could be implicitly constructed from single-element initializer lists
- `makeFlatVector(const std::vector<T>& data, ...)` - which accepts vector-like containers

This ambiguity made it difficult for users to create simple flat vectors using the convenient brace initialization syntax, which is a common pattern in test code.

## Solution
Added a new template overload that explicitly accepts `std::initializer_list<T>`:
```cpp
template <typename T>
FlatVectorPtr<EvalType<T>> makeFlatVector(
    const std::initializer_list<T>& data,
    const TypePtr& type = CppToType<T>::create())
```

## Benefits
- Resolves compilation ambiguity when using brace initialization
- Allows users to write more concise test code: `makeFlatVector<int64_t>({1})`
- Maintains backward compatibility with existing code
- Follows the same pattern as the existing vector-based overload


The test code is output like this when it is not modified:
<img width="433" height="154" alt="Screenshot 2025-08-19 at 20 01 20" src="https://github.com/user-attachments/assets/d202844b-d8d7-488b-9143-267f59532ea0" />

mbasmanova commented on velox/vector/tests/utils/VectorTestBase.h:
```diff
@@ -195,6 +195,16 @@ class VectorTestBase {
     return vectorMaker_.flatVector<T>(data, type);
   }
 
+  // Helper overload to allow users to use initializer list directly without
```
Perhaps, simplify this comment to show what it enables.

```
// Enables makeFlatVector({123}).
```

mbasmanova commented on velox/vector/tests/VectorTest.cpp:
```diff
@@ -1609,6 +1609,19 @@ TEST_F(VectorTest, rowPrepareForReuse) {
 }
 
+TEST_F(VectorTest, makeFlatVectorWithInitializerList) {
+  // Test makeFlatVector basic functionality with initializer_list
```
These comments are not needed. Let's remove.

mbasmanova commented on velox/vector/tests/VectorTest.cpp:
```diff
@@ -1609,6 +1609,19 @@ TEST_F(VectorTest, rowPrepareForReuse) {
+  // This should now work correctly with the added initializer_list overload
+  auto vector = makeFlatVector<int64_t>({803});
+
+  // Verify the vector is not null and has the expected size
```
redundant comment; let's remove

mbasmanova commented on velox/vector/tests/VectorTest.cpp:
```diff
@@ -1609,6 +1609,19 @@ TEST_F(VectorTest, rowPrepareForReuse) {
+  ASSERT_NE(vector, nullptr);
+  EXPECT_EQ(vector->size(), 1);
+
+  // Verify the value at index 0 is 803, not 0
```
ditto

mbasmanova reviewed (commented):
@Super-long Thank you for the enhancement. It is important to allow easy to read and write tests.

mbasmanova reviewed (approved):
Thanks!

Super-long commented:
> @Super-long CI is red. Would you take a look?

Of course, this is a test code problem, which seems unrelated to this change. I made the following changes. Is it necessary to separate the two changes?

## Problem

The original code defined `overflowSize` as:
```cpp
static constexpr int64_t overflowSize = 3UL << 30;
```

This caused a compilation error on macOS with the message:
```
implicit conversion from 'const int64_t' (aka 'const long long') to 'vector_size_t' (aka 'int') changes value from 3221225472 to -1073741824 [-Werror,-Wconstant-conversion]
```

The issue occurred because `3UL << 30` equals 3,221,225,472, which exceeds the maximum value for a 32-bit signed integer (2,147,483,647). When implicitly converted to `vector_size_t` (which is `int`), it caused an integer overflow.

## Solution

### 1. Replaced the overflow-prone value with a simple negative number:
```cpp
// Before
static constexpr int64_t overflowSize = 3UL << 30;

// After  
static constexpr vector_size_t invalidSize = -1;
```

### 2. Renamed the variable for better clarity:
- Changed `overflowSize` to `invalidSize` throughout the file (16 occurrences)
- The new name better reflects the actual purpose: testing invalid size handling

## Benefits

- **Simpler and clearer:** Using `-1` is more straightforward than dealing with integer overflow
- **Type safe:** Directly uses `vector_size_t` type, avoiding implicit conversions
- **Same test coverage:** Still effectively tests the `VELOX_CHECK_GE(size, 0)` validation in `ensureSize()` method
- **Better semantics:** `invalidSize` clearly indicates this is testing invalid input handling
- **Cross-platform compatibility:** Works consistently across different architectures

## How it works

The test calls `writer.ensureSize(invalidSize)` where `invalidSize = -1`. The `ensureSize()` method contains `VELOX_CHECK_GE(size, 0)` which fails when `size = -1`, throwing an exception that is caught and verified by `VELOX_ASSERT_THROW`.

Super-long commented:
> @Super-long Thanks. Curious, are you using AI to write these comments / descriptions?

hhh, Yes, I wrote the code and used AI to write reviews using the Problem, Solution, Benefits framework.
